dimod/binary/cybqm/cybqm_float64.pxd

```diff
@@ -0,0 +1,28 @@
+# distutils: language = c++
+# cython: language_level=3
+
+# Copyright 2021 D-Wave Systems Inc.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+cimport numpy as np
+
+ctypedef np.float64_t bias_type
+ctypedef np.int64_t index_type
```

Comment:
Why do we need the type of index to be of 64 bit ? There will be quite a bit of wastage of memory.

Reply:
I got this from you! :laughing: I must have misunderstood some benchmarks you sent me. I thought it was odd that it would matter they two different vectors have items of the same length.

Reply:
Okay if they have same length data type the performance will be better lets say 20% better for getting the effective biases from the CQM , it is true. Getting efffective bias is all that matters for binary and more or less for discrete sampling. For integers, it is the sampling that dominates. Lets say in total we will have 15% better performance if bias_type and index_type have same width. But as I understand 30% storage is more important than 15% performance in our case. If it is not the case we can definitely keep them of the same width :) 

If the variables are all of discrete type this will take up more than 30% of the space as I explained earlier for discrete variables we have two extra vectors of index_type. This is your call here :) . If performance is the goal we should keep this.

Reply:
No, you're right, in this case it's memory that we care about. I will make the change (luckily it's super easy to do).

Reply:
FYI, when I gather the 4 double precision biases, I have to supply 4 long long int indices to the cpu , so if they are of type int32_t  I need to convert them to int64_t that conversion takes up that extra 20% time in the tiny loop.

Reply:
Okay this discussion was fruitful I had missed one instruction that can actually gather without the conversion I will try it out and in that case we may save space and still have some performance :) 

https://software.intel.com/sites/landingpage/IntrinsicsGuide/#text=gather&techs=AVX,AVX2&expand=2999